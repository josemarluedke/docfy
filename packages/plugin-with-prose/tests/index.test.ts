import Docfy from '@docfy/core';
import path from 'path';
import withProse from '../src';

const root = path.resolve(__dirname, './__fixtures__');

test('it should work', async () => {
  const docfy = new Docfy({
    plugins: [withProse],
  });
  const result = await docfy.run([
    {
      root,
      urlPrefix: 'docs',
      pattern: '**/*.md',
    },
  ]);

  const htmls = [];
  result.content.forEach(page => {
    htmls.push([page.source, page.rendered]);

    page.demos?.forEach(demo => {
      htmls.push([demo.source, demo.rendered]);
    });
  });

  expect(htmls).toMatchSnapshot();
});

test('it should use custom class', async () => {
  const docfy = new Docfy({
    plugins: [withProse({ className: 'prose dark:prose-invert' })],
  });
  const result = await docfy.run([
    {
      root,
      urlPrefix: 'docs',
      pattern: '**/*.md',
    },
  ]);

  const htmls = [];
  result.content.forEach(page => {
    htmls.push([page.source, page.rendered]);

    page.demos?.forEach(demo => {
      htmls.push([demo.source, demo.rendered]);
    });
  });

  expect(htmls).toMatchSnapshot();
});
