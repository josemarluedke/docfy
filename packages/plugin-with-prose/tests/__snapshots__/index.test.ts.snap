// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should use custom class 1`] = `
Array [
  Array [
    "README.md",
    "<div class=\\"prose dark:prose-invert\\">
<h1 id=\\"the-index\\">The index</h1>
<p>Some text</p>
<div>
  This is a DIV
</div>
<p>Some other text</p>
<div class=\\"not-prose additional classes\\">
<ul>
<li>this list</li>
<li>has</li>
<li>escaped prose's CSS</li>
</ul>
</div>
</div>",
  ],
  Array [
    "button.md",
    "<div class=\\"prose dark:prose-invert\\">
<h1 id=\\"button-component\\">Button Component</h1>
<p>Some text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-hbs\\">&#x3C;button>This is Preview Templace&#x3C;/button>
</code></pre>
</div>
</div>",
  ],
  Array [
    "button-demo/demo1.md",
    "<div class=\\"prose dark:prose-invert\\">
<h1 id=\\"button-demo\\">Button demo</h1>
<p>Some text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-html\\">&#x3C;button>Cool&#x3C;/button>
</code></pre>
</div>
<p>Some other text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-html\\">&#x3C;button>Cool&#x3C;/button>
</code></pre>
</div>
<p>More text</p>
<pre><code class=\\"language-sh\\">ember install @docfy/ember
</code></pre>
</div>",
  ],
]
`;

exports[`it should work 1`] = `
Array [
  Array [
    "README.md",
    "<div class=\\"prose\\">
<h1 id=\\"the-index\\">The index</h1>
<p>Some text</p>
<div>
  This is a DIV
</div>
<p>Some other text</p>
<div class=\\"not-prose additional classes\\">
<ul>
<li>this list</li>
<li>has</li>
<li>escaped prose's CSS</li>
</ul>
</div>
</div>",
  ],
  Array [
    "button.md",
    "<div class=\\"prose\\">
<h1 id=\\"button-component\\">Button Component</h1>
<p>Some text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-hbs\\">&#x3C;button>This is Preview Templace&#x3C;/button>
</code></pre>
</div>
</div>",
  ],
  Array [
    "button-demo/demo1.md",
    "<div class=\\"prose\\">
<h1 id=\\"button-demo\\">Button demo</h1>
<p>Some text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-html\\">&#x3C;button>Cool&#x3C;/button>
</code></pre>
</div>
<p>Some other text</p>
<div class=\\"not-prose\\">
<pre><code class=\\"language-html\\">&#x3C;button>Cool&#x3C;/button>
</code></pre>
</div>
<p>More text</p>
<pre><code class=\\"language-sh\\">ember install @docfy/ember
</code></pre>
</div>",
  ],
]
`;
